src/workerd/io/worker.c++

```diff
@@ -449,37 +449,84 @@ public:
       // We're on a request-serving thread.
       auto& ioContext = IoContext::current();
       timePoint = ioContext.now();
-    } else KJ_IF_MAYBE(info, inspectorTimerInfo) {
-      if (info->threadId == getCurrentThreadId()) {
-        // We're on an inspector-serving thread.
+    } else {
+      auto lockedState = state.lockExclusive();
```

Comment:
Dues this ever get called recursively within `runMessageLoopOnPause()`? E.g. if a message was delivered indicating the caller wanted to invoke `console.time()` directly from the JS console?

Granted, _usually_ `IoContext::hasCurrent()` should be true there, but what if a breakpoint was set in global startup where there is no IoContext?

Reply:
I've not seen a case where we recurse into this.

Devtools proper on regular web pages and ew fiddle behave the same way: `console.time()` returns "undefined" when it is called from the console; the second time it is invoked it also logs a warning: `Timer 'default' already exists`.

There is a potential `ew-bin-test` to follow. It might be good to have a workerd test too. If we could launch isolates within a config in separate threads, then we could have a wd-test with two isolates: a worker to be tested and another that sends CDP messages to it to exercise functionality.